osu.Game/Overlays/BeatmapListing/BeatmapSearchScoreFilterRow.cs

```diff
@@ -31,18 +33,36 @@ public RankItem(ScoreRank value)
             {
             }
 
-            protected override string LabelFor(ScoreRank value)
+            protected override LocalisableString LabelFor(ScoreRank value)
             {
                 switch (value)
                 {
                     case ScoreRank.XH:
-                        return @"Silver SS";
+                        return BeatmapsStrings.RankXH;
+
+                    case ScoreRank.X:
+                        return BeatmapsStrings.RankX;
 
                     case ScoreRank.SH:
-                        return @"Silver S";
+                        return BeatmapsStrings.RankSH;
+
+                    case ScoreRank.S:
+                        return BeatmapsStrings.RankS;
+
+                    case ScoreRank.A:
+                        return BeatmapsStrings.RankA;
+
+                    case ScoreRank.B:
+                        return BeatmapsStrings.RankB;
+
+                    case ScoreRank.C:
+                        return BeatmapsStrings.RankC;
+
+                    case ScoreRank.D:
+                        return BeatmapsStrings.RankD;
 
                     default:
-                        return value.GetDescription();
+                        throw new ArgumentException("Unsupported value.", nameof(value));
```

Comment:
Bit wary of throwing in a function which could cause a crash of the game from an incorrect external (api) value.

Reply:
It's not actually an external API value in this case. These are being instantiated via

https://github.com/ppy/osu/blob/7654df94f6f37b8382be7dfcb4f674e03bd35427/osu.Game/Overlays/BeatmapListing/BeatmapSearchFilterRow.cs#L83-L84

i.e. all items here should already be defined in the enum. I'd say it's *probably* fine to crash on that, but for ranks `ToString()` may also be an ok default.

Reply:
Right. Just thinking as a general rule though, we probably want to stop throwing on these kinds of lookups as it's more likely to cause issues than help.

Reply:
I would rather things throw than silently fail? If we ever add an SSSH or something like that.

Reply:
Well consider the case we forget to open beatmap listing and it somehow doesn't throw in a test and makes it into a release: we either have users reporting "why is SSSH showing as 19?" or "OH MY GOD BEATMAP LISTING IS CRASHING THE GAME ON OPENING".

food for thought anyways, i don't have any strong opinion for now.